cpp/tests/prims/mg_copy_v_transform_reduce_inout_nbr.cu

```diff
@@ -0,0 +1,506 @@
+      handle.get_comms().barrier();
+      double elapsed_time{0.0};
+      hr_clock.stop(&elapsed_time);
+      std::cout << "MG copy v transform reduce in took " << elapsed_time * 1e-6 << " s.\n";
+    }
+
+    if (cugraph::test::g_perf) {
+      CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      handle.get_comms().barrier();
+      hr_clock.start();
+    }
+
+    copy_v_transform_reduce_out_nbr(
+      handle,
+      mg_graph_view,
+      row_prop.device_view(),
+      col_prop.device_view(),
+      [] __device__(auto row, auto col, weight_t wt, auto row_property, auto col_property) {
+        if (row_property < col_property) {
+          return row_property;
+        } else {
+          return col_property;
+        }
+      },
+      property_initial_value,
+      cugraph::get_dataframe_buffer_begin(out_result));
+
+    if (cugraph::test::g_perf) {
+      CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      handle.get_comms().barrier();
+      double elapsed_time{0.0};
+      hr_clock.stop(&elapsed_time);
+      std::cout << "MG copy v transform reduce out took " << elapsed_time * 1e-6 << " s.\n";
+    }
+
+    //// 4. compare SG & MG results
+
+    if (prims_usecase.check_correctness) {
+      cugraph::graph_t<vertex_t, edge_t, weight_t, store_transposed, false> sg_graph(handle);
+      std::tie(sg_graph, std::ignore) =
+        cugraph::test::construct_graph<vertex_t, edge_t, weight_t, store_transposed, false>(
+          handle, input_usecase, true, false);
+      auto sg_graph_view = sg_graph.view();
+
+      auto sg_vertex_property_data = generate<result_t>::vertex_property(
+        thrust::make_counting_iterator(sg_graph_view.get_local_vertex_first()),
+        thrust::make_counting_iterator(sg_graph_view.get_local_vertex_last()),
+        hash_bin_count,
+        handle);
+      auto sg_col_prop =
+        generate<result_t>::column_property(handle, sg_graph_view, sg_vertex_property_data);
+      auto sg_row_prop =
+        generate<result_t>::row_property(handle, sg_graph_view, sg_vertex_property_data);
+      result_compare comp{handle};
+
+      auto global_out_result = cugraph::allocate_dataframe_buffer<property_t>(
+        sg_graph_view.get_number_of_local_vertices(), handle.get_stream());
+      copy_v_transform_reduce_out_nbr(
+        handle,
+        sg_graph_view,
+        sg_row_prop.device_view(),
+        sg_col_prop.device_view(),
+        [] __device__(auto row, auto col, weight_t wt, auto row_property, auto col_property) {
+          if (row_property < col_property) {
+            return row_property;
+          } else {
+            return col_property;
+          }
+        },
+        property_initial_value,
+        cugraph::get_dataframe_buffer_begin(global_out_result));
+      auto expected_out_result =
+        permute_result(handle, global_out_result, (*d_mg_renumber_map_labels));
```

Comment:
Can't we use cugraph::test::sort_by_key (https://github.com/rapidsai/cugraph/blob/branch-21.10/cpp/tests/centrality/mg_katz_centrality_test.cpp#L148) instead (so no need to reinvent the wheel)?

Reply:
I will need to change sort_by_key to accept dataframe iterators instead of pointers. And it will need to be converted into a header only templated function since we cannot possible specialize for every tuple type we need for testing. Was the motivation behind specializing test::sort_by_key to reduce compile time? Is it a deal breaker if its not specialized anymore? or do you want to specialize it for a select few tuple types and restrict testing for those types in mg tests?

Reply:
> I will need to change sort_by_key to accept dataframe iterators instead of pointers. And it will need to be converted into a header only templated function since we cannot possible specialize for every tuple type we need for testing. Was the motivation behind specializing test::sort_by_key to reduce compile time? Is it a deal breaker if its not specialized anymore? or do you want to specialize it for a select few tuple types and restrict testing for those types in mg tests?

The key motivation was to compile with g++ instead of requiring nvcc. This doesn't apply for primitives but for our C++ algorithms, we want to make sure they are useable with GCC. Let me think more about how we should handle this.

Reply:
> do you want to specialize it for a select few tuple types and restrict testing for those types in mg tests?

Yeah... I think is this is a reasonable solution considering that this is only for testing. We may revisit this issue if we see more use cases for this (so specializing for every used types becomes to messy) later.

Reply:
Maybe later... if we really need to provide this as a feature to users, we need to implement this by something like
```
convert std::tuple of rmm::device_uvector to std::vector of pointers

  for (size_t i = 0; i < tuple_size; ++i) {
    if (sizeof tuple element type == 1) {
      shuffle in a type agnostic way.
    }
    ...
    else if (sizeof tuple element type == 8) {
      shuffle in a type agnostic way.
    }
    else {
      CUGRAPH_FAIL(...);
    }
  }
```
But this sounds like a little too much work right at this moment.